packages/core/saved-objects/core-saved-objects-server/src/saved_objects_type.ts

```diff
@@ -70,6 +70,7 @@ export interface SavedObjectsType<Attributes = any> {
   mappings: SavedObjectsTypeMappingDefinition;
   /**
    * An optional map of {@link SavedObjectMigrationFn | migrations} or a function returning a map of {@link SavedObjectMigrationFn | migrations} to be used to migrate the type.
+   * @deprecated Use {@link SavedObjectsType.modelVersions | modelVersions} instead.
```

Comment:
Deprecates legacy properties to make it clear that going forward, changes have to be defined with new model versions.